bottlecap/src/traces/stats_concentrator.rs

```diff
@@ -1,38 +1,216 @@
 use crate::config::Config;
 use datadog_trace_protobuf::pb;
-use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+use std::{collections::HashMap, sync::Arc};
+use tracing::error;
 
 // Event sent to the stats concentrator
-#[derive(Clone, Copy)]
+#[derive(Clone)]
 pub struct StatsEvent {
     pub time: u64,
     pub aggregation_key: AggregationKey,
     pub stats: Stats,
 }
 
-#[derive(Clone, Debug, PartialEq, Eq, Hash, Copy)]
-pub struct AggregationKey {}
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct AggregationKey {
+    pub env: String,
+    pub service: String,
+    // e.g. "aws.lambda.load", "aws.lambda.import"
+    pub name: String,
+    // e.g. "my-lambda-function-name", "datadog_lambda.handler", "urllib.request"
+    pub resource: String,
+    // e.g. "aws.lambda.load", "aws.lambda.import"
```

Comment:
```suggestion
    // e.g. "serverless"
```

not sure if the comment is correct for `r#type`

Reply:
You are right. Will fix.